### PR TITLE
TypeVariableName.withBounds added

### LINKED ISSUE #440: Add TypeVariableName.withBounds
I use a constant for type variables and I want to add bounds when using in a type declaration.

e.g.,
```java
TypeSpec.classBuilder("Example")
    .addTypeVariable(tVar.withBounds(boundTypeName))
    .addField(tVar, "whatever", PRIVATE FINAL)
```

### PR DESCRIPTION
**Pull request by sormuras**
Closes #440

### CODE REVIEW

**Comment by JakeWharton** [comment]
Test?

File: src/main/java/com/squareup/javapoet/TypeVariableName.java [hunk 1]
```diff
@@ -57,6 +57,18 @@ private TypeVariableName(String name, List<TypeName> bounds, List<AnnotationSpec
     return new TypeVariableName(name, bounds);
   }
 
+  public TypeVariableName withBounds(Type... bounds) {
+    return get(name, bounds);
```
Comment (JakeWharton) [inline src/main/java/com/squareup/javapoet/TypeVariableName.java]: This loses any existing bounds and annotations. I think you'll want to combine just like `annotated` does.

Comment (sormuras) [inline src/main/java/com/squareup/javapoet/TypeVariableName.java]: Prepend or append new bounds?

Comment (JakeWharton) [inline src/main/java/com/squareup/javapoet/TypeVariableName.java]: Append, but I don't think it matters. They should probably be a Set in actuality.

Comment (tbroyer) [inline src/main/java/com/squareup/javapoet/TypeVariableName.java]: Aren't bounds supposed to have a class first (if any) followed by interfaces?
https://docs.oracle.com/javase/specs/jls/se7/html/jls-4.html#jls-4.4

**Comment by sormuras** [comment]
Test! :smile:

File: src/main/java/com/squareup/javapoet/TypeVariableName.java (hunk 1 above)
Comment (JakeWharton) [inline src/main/java/com/squareup/javapoet/TypeVariableName.java]: True. We always do order-preserving sets, but I was thinking of a case where you're potentially building up interface-based bounds. Might as well leave it as a list until someone complains with an actual use case.

**Comment by swankjesse** [comment]
Thanks!

**Pull request merged**